guides/v2.1/contributor-guide/backward-compatible-development/index.md

```diff
@@ -39,6 +39,12 @@ Mark the method with the `@deprecated` tag instead of removing it.
 
 Continue returning the same results from the method if possible, so the old functionality is preserved.
 
+If you need to remove the existing functionality and you do not need alternative implementation, you SHOULD change the method code to
```

Comment:
I'd rephrase, "and you do not need alternative implementation" to "and there is no alternative implementation" because developer of this part of code obviously don't need this functionality, but we need to care about other developers, and they may need it. So usually we should preserve the functionality even if core Magento doesn't need it. The only exception is when it's impossible to provide an alternative implementation.